app/src/main/java/doit/study/droid/QuestionFragment.java

```diff
@@ -44,16 +46,19 @@
     private LinearLayout mvAnswersLayout;
     private TextView mvRight;
     private TextView mvWrong;
+    private Button mvNextButton;
 
 
     // Factory method
-    public static QuestionFragment newInstance(int position) {
+    public static QuestionFragment newInstance(int position, int questionId, AnswerCheckListener listener) {
         if (DEBUG) Log.i("NSA", "newInstance "+position);
         // add Bundle args if needed here before returning new instance of this class
         QuestionFragment fragment = new QuestionFragment();
         Bundle bundle = new Bundle();
         bundle.putInt(ID_KEY, position);
         fragment.setArguments(bundle);
+        fragment.mQuestionId = questionId;
+        fragment.mAnswerCheckListener = listener;
```

Comment:
By the way, I like the "next button" mechanics, but if a user wants to try one more time, he has to swipe back.
We can make setting auto-swipe or next button mode.